src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlClientEventSource.cs

```diff
@@ -924,148 +893,111 @@ internal void EndExecute(int objectId, int compositeState, int sqlExceptionNumbe
         }
 
         [Event(TraceEventId, Level = EventLevel.Informational, Keywords = Keywords.Trace)]
-        internal void Trace(string message)
-        {
+        internal void Trace(string message) =>
             WriteEvent(TraceEventId, message);
-        }
 
         [Event(ScopeEnterId, Level = EventLevel.Informational, Opcode = EventOpcode.Start, Keywords = Keywords.Scope)]
         internal long ScopeEnter(string message)
         {
             long scopeId = Interlocked.Increment(ref s_nextScopeId);
-            WriteEvent(ScopeEnterId, message);
+            WriteEvent(ScopeEnterId, string.Format(message, scopeId));
```

Comment:
It's required an external consideration; It's better to manage it completely here and add it to the input `message` to avoid the possible run time exception.
- same for the similar implementations inside this class.